pkg/cluster/cluster.go

```diff
@@ -468,6 +481,66 @@ func (c *Cluster) Create() (err error) {
 	return nil
 }
 
+func (c *Cluster) createPgbackrestRepoHostObjects(repo cpov1.Repo, first_repo bool) error {
+	c.setProcessName("Creating pgbackrest repo-host related objects")
+	c.logger.Info("Creating pgbackrest repo-host related objects")
+
+	pvcName := c.getPgbackrestRepoHostName()
+	backupSecretName := c.getPgbackrestCertSecretName()
+	if first_repo {
+		if err := c.createPgbackrestRepoHostConfig(); err != nil {
+			err = fmt.Errorf("could not create a pgbackrest repo-host config: %v", err)
+			return err
+		}
+	}
+	// if statefulset for pvc is already created then only mount an additional volume in it for this repo
+	if !first_repo {
+		c.logger.Info("need more pvc based pgbackrest repo-host, hence deleting old sts")
+		if err := c.KubeClient.StatefulSets(c.Namespace).Delete(context.TODO(), c.getPgbackrestRepoHostName(), metav1.DeleteOptions{}); err != nil {
```

Comment:
I don't get this line, if we have more than one repository we just delete the only one we have.

Reply:
Collecting all the info we need and then creating seems way cleaner. Right now the logic is very confusing. From what I can tell it seems that on the first iteration we create the repo statefulset, but only if the repo secret does not already exist, and if we run a second iteration we delete the stateful set but do not recreate anything.